### PR TITLE
[Feat] File rename support

### PR DESCRIPTION
**Pull request by rutvik110**
[Feat] File rename support

Adds support to send file rename notifications to LSP.

### CODE REVIEW

File: example/lib/lsp_exploration/lsp/messages/initialize.dart [hunk 1]
```diff
@@ -133,8 +133,10 @@ class LspClientCapabilities {
             'documentChanges': true,
             'resourceOperations': ['create', 'rename', 'delete']
           },
+          "applyEdit": true,
```
Comment (angelosilvestre) [inline example/lib/lsp_exploration/lsp/messages/initialize.dart]: We already have "applyEdit" set above. This is probably a merge conflict with my latest merged PR.

File: example/lib/ide/ide.dart [hunk 2]
```diff
@@ -315,7 +346,9 @@ class BottomBar extends StatelessWidget {
             const SizedBox(width: 8),
             // This is here to experiment with the LspClient
             GestureDetector(
-              onTap: _renameFile,
+              onTap: () {
+                _renameFile(context);
+              },
```
Comment (angelosilvestre) [inline example/lib/ide/ide.dart]: Can we have the rename be invoked by pressing ENTER on a file selected in the file tree instead of this button?

Comment (rutvik110) [reply example/lib/ide/ide.dart]: Sure thing. I'll update it.

**Comment by rutvik110** [comment]
@angelosilvestre Made the requested updates and updated the rename file flow to launch rename file dialog when pressed enter on a file in the tree view.

**Comment by rutvik110** [comment]
@angelosilvestre Just pinging you again in case you missed this.

File: example/lib/ide/file_explorer/file_explorer.dart [hunk 3]
```diff
@@ -93,6 +104,37 @@ class _FileExplorerState extends State<FileExplorer> {
     }
   }
 
+  Future<void> _renameFile(BuildContext context, String currentFilePath) async {
+    print("Rename file");
```
Comment (angelosilvestre) [inline example/lib/ide/file_explorer/file_explorer.dart]: Please remove the print statements.

File: example/lib/ide/file_explorer/file_explorer.dart [hunk 4]
```diff
@@ -93,6 +104,37 @@ class _FileExplorerState extends State<FileExplorer> {
...
+    final newFileName = await showDialog<String>(
+        context: context,
+        builder: (context) {
+          return const RenameFileDialog();
+        });
```
Comment (angelosilvestre) [inline example/lib/ide/file_explorer/file_explorer.dart]: We should handle the case where the user does not input the new file name.

Comment (rutvik110) [reply example/lib/ide/file_explorer/file_explorer.dart]: Updated.

**Review by angelosilvestre — APPROVED** [review]
LGTM

**Pull request merged**